core/src/test/java/org/apache/calcite/test/RexProgramTest.java

```diff
@@ -1740,6 +1731,19 @@ private void checkExponentialCnf(int n) {
         "true");
   }
 
+  @Test public void testSimplifyItemRangeTerms() {
+    RexNode item = item(input(tArray(tInt()), 3), literal(1));
+    simplify = this.simplify.withParanoid(false);
```

Comment:
What is the point of this line?
If it is not required please remove it. If it is required please add a clarification comment.

Reply:
The validation simply doesn't support array datatype and I didn't find right way to add it.